swiftide-integrations/src/openai/mod.rs

```diff
@@ -211,6 +213,66 @@ impl<C: async_openai::config::Config + Default> GenericOpenAI<C> {
     }
 }
 
+pub fn openai_error_to_language_model_error(e: OpenAIError) -> LanguageModelError {
```

Comment:
I suspect that if you replace this with a `impl From<OpenAIError> for LanguageModelError`, you can remove every single line with `.map_err(openai_error_to_language_model_error)` as along as there is a `?` at the end.

Reply:
There's also some other places where there's map_err's but the type of error is always permanent (i.e. io errors). Potential to remove a lot of code everywhere.

Reply:
I can't `impl From<OpenAIError> for LanguageModelError` without adding `async_openai` as an (optional) dependency for swiftide core, or do you see another way?

Reply:
Huh yeah, really don't want that... Maybe  make the error associated in the trait and put it in integrations, or use an owned type pattern so you can do the impl in integrations, or approach the problem differently. Probably more ways. I don't know if it's worth going through all those hoops just to try this out. It would remove a lot of code though, so there's a clear upside there, up to you 🤷